archiver/main.go

```diff
@@ -265,6 +269,52 @@ func getAllCampaigns(config LambdaConfig) ([]KnowBe4FlatCampaign, error) {
 	return flatResults, err
 }
 
+func getGroupsPage(pageNum int, config LambdaConfig) ([]KnowBe4Group, error) {
+	queryParams := map[string]string{
+		"per_page": strconv.Itoa(countPerPage),
+		"page":     strconv.Itoa(pageNum),
+	}
+
+	// Make http call
+	resp, err := callAPI(groupsURLPath, config, queryParams)
+	if err != nil {
+		return nil, err
+	}
+
+	defer resp.Body.Close()
+	bodyBytes, _ := ioutil.ReadAll(resp.Body)
+
+	var groups []KnowBe4Group
+
+	if err := json.Unmarshal(bodyBytes, &groups); err != nil {
+		return nil, fmt.Errorf("error decoding response json for groups: %s", err)
+	}
+
+	return groups, nil
+}
+
+func getAllGroups(config LambdaConfig) ([]KnowBe4FlatGroup, error) {
+	var allGroups []KnowBe4Group
+
+	for i := 1; ; i++ {
+		c, err := getGroupsPage(i, config)
+		if err != nil {
```

Comment:
I'm curious what happens if a page is full, but there aren't any more pages afterwards, will this throw an error?

Reply:
Good question. I just tried with Insomnia, giving a page number beyond the end of the range of data and it responded with 200 and an empty list.